### PR TITLE
Create form for adding new users to the application

### PR DESCRIPTION
**Pull request by vikitaotiz**
The project tests the use of forms in rails

### CODE REVIEW

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    <div class="form-group mb-3">
+      <%= form.label :password, class: "form-label"  %>
+      <%= form.text_field :password, type: 'password', class: "form-control" %>
+    </div>
+
+    <div class="form-group d-grid">
+      <%= form.submit nil, class: 'btn btn-success btn-sm' %>
+    </div>
+  <% end %>
```
Comment (BrianSammit) [inline app/views/users/new.html.erb]: 1. In line 9 instead of `text_field` use `email_field`
2. In line 14 instead of `text_field` use `password_field`

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :red_circle: :stop_sign: 
--------------
Hello @vikitaotiz great job you did so far in this project :+1: :100: 
TO HIGHLIGHT: 

- Your repo is professional
- Your  project is professional
- I like how your app looks 
- Great job for the extra effort with the design

However, there are a few changes to make to complete this milestone :hammer_and_wrench: see them below :eyes: :point_down: 
-------------
- [ ] You must be able to edit a user. Right now the app doesn't have the action to edit. Please add it. 

edit_user

- [ ] Please add the `:edit, :update` actions in the routes. 

user_route2

- [ ] On create action in the user controller it should redirect to `new_user_path`.

user_create

- [ ] Please create the edit and update action on your user controller. 

user_route2 (1)

- [ ] Please remove the files on your views that you are not using. In this case, is the `create.html.erb`
- [ ] Please add the commented-out forms. Don't delete the first two forms that you created just leave them in the code commented. 
- [ ] Chek my inline comments. :point_down: 
--------------
Make these required changes and send it again for another review :muscle: :+1: 
Reach me on SLACK if you have any questions. 
@ Brian Sammit Cruz Rdoriguez
HAPPY CODING!!!!

**Comment by rukundoeric** [comment]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - Follows gitwork-flow.  💯
  - Codebase looks clean and professional. 💯 
  - Implemented some of the feedback raised from the previous review. 💯 


- ### To Improve
  - [ ] [MANDATORY] As requested by the previous reviewer, two versions of the form you were asked to implement before in the project specifications are not to be deleted but commented out. kindly, go in `app/views/user/_form.html.erb` and bring them back and comment them out.

_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by MkrtichSargsyan — CHANGES_REQUESTED** [review]
## _Status:_ Changes Requested :recycle:

## Hello Guys

THIS IS VERY IMPRESSIVE WORK. YOU HAVE DONE A GOOD JOB :+1:

## To Highlight :tada:

1. The project solves the problem it was supposed to solve.
2. Your PR has a good title and description.
3. You have descriptive git commit messages.


Good job on implementing the changes requested. However, there are some changes that you need to make in order to get your project approved

 - [ ] You were asked to implement two versions of the form 
 Kindly also add ```form_tag``` and the ```regular form``` and comment them out

          ex. 1
      
          <form action="users_path" method="post" accept-charset="UTF-8">
            <label for="user[username]">Username</label>
            <input name="user[username]" type="text" />
            <input
              type="hidden"
              name="authenticity_token"
              value="<%= form_authenticity_token %>"
            />
            <input type="submit" value="Submit" />
          </form>
      
          ex. 2
      
          <%= form_tag(users_path) do %> <%= label_tag 'username' %> <%= text_field_tag
            'username' %> <%= label_tag 'email' %> <%= email_field_tag 'email' %> <%=
            label_tag 'password' %> <%= password_field_tag 'password' %> <%= submit_tag %>
            <% end %>


<hr>

Please, make the REQUIRED changes and submit your request again! :grinning:
Good luck. 

> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

File: app/views/users/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+     <label for="username" class="form-label">Username : </label>
+     <input name="username" type="text" class="form-control" />
+     <label for="email" class="form-label">Email : </label>
+     <input name="email" type="text" class="form-control" />
+     <label for="password" class="form-label">Password : </label>
+     <input name="password" type="password" class="form-control" />
+     <input type="submit" value="Submit" class="btn btn-primary" />
+   </form>
+
+<% end %>
```
Comment (SajjadAhmad14) [inline app/views/users/_form.html.erb:13]: [OPTIONAL] Please remove logic from views. You can use helper under app/helpers to put your logic in and render it in views as you need.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [ ] You have mastered Git flow.
- [ ] App runs perfectly.
## To Improve 🙏
- [ ] Code of HTML form and that of form created with form_tag should be commented out. app/views/users/new.html.erb should only have active code of form_with. Kindly refer to the requirements in the Odin project again and implement changes.

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏

### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Pull request merged**